docs/application/dotnet/guides/nui/grid-layout.md

```diff
@@ -1,4 +1,112 @@
+## Columns and rows
+
+The `Columns` and the `Rows` properties specify the maximum number of cells on a horizontal or vertical axis.
+
+When the `Column` or the `Row` of child is automatically set without specified value, the value cannot be exceeded the number of main-axis cell which is the `Columns` value on horizontal orientation or the `Rows` value on vertical orientation. But, it can be assigned a value that exceeds the number of cross-axis cell.
+
+The following figure shows how to set the `Colomns` and the `Rows` properties. Since the `GridOrientation` is set to horizontal, the row index of last item can be `2` that exceeds the `Rows` property.
+
+![Column](./media/columnLayout.png)
+
+```csharp
+View layoutView = new View();
+var gridLayout = new GridLayout();
+gridLayout.GridOrientation = GridLayout.Orientation.Horizontal;
+gridLayout.Columns = 2;
+gridLayout.Rows = 1;
+layoutView.Layout = gridLayout;
+```
+<a name="grid-orientation"></a>
+## Grid orientation
+
+The `GridOrientation` only works when a child is added without the specified `Row` or `Column` property.
+
+ - On horizontal orientation, the children will automatically occupy empty cells from left to right.
+ - On vertical orientation, the children will automatically occupy empty cells from top to bottom.
+
+The following figure shows how the `GridOrientation` property works:
+
+![gridOrientation](./media/gridOrientation.png)
+
+## Column spacing and row spacing
+
+The `ColumnSpacing` and `rowSpacing` properties are the distance between columns or rows. In order to set same sapce between the children, you can use `ColumnSpacing` and `rowSpacing` properties instead of setting the same margin for each child.
+
+The following figure shows how the `ColumnSpacing` and `RowSpacing` properties work:
+
+![gridSpacing](./media/gridSpacing.png)
+
+The following table describes the properties that affect the specified child:
+
+| Property               | Type            | Description |
+| -----------------------| --------------- | ----------- |
+| `Column`               | int             | Gets or sets the column index of the child. |
+| `Row`                  | int             | Gets or sets the row index of the child. |
+| `ColumnSpan`           | int             | Gets or sets the number of columns that the child occupy. |
+| `RowSpan`              | int             | Gets or sets the number of rows that the child occupy. |
+| `HorizontalStretch`    | StretchFlags    | Gets or sets the value how child is resized within horizontal space of the cell where child occupy. |
+| `VerticalStretch`      | StretchFlags    | Gets or sets the value how child is resized within vertical space of the cell where child occupy. |
+| `HorizontalAlignment`  | Alignment       | Gets or sets the horizontal alignment of the child within the cell. |
+| `VerticalAlignment`    | Alignment       | Gets or sets the vertical alignment of the child within the cell. |
+
+## Column and row
```

Comment:
Some example related to this property might be useful, just my opinion.

Reply:
Added :)